### PR TITLE
[BEAM-646] Add ParDo Matchers to PTransformMatchers

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---
These match Splittable ParDos and ParDos that use State and Timers.

Update tests to remove excess generic args.

### CODE REVIEW

**Comment by tgroh** [comment]
R: @kennknowles

File: runners/core-java/src/main/java/org/apache/beam/runners/core/PTransformMatchers.java [hunk 1]
```diff
@@ -56,4 +60,64 @@ public boolean matches(AppliedPTransform<?, ?, ?> application) {
       return application.getTransform().getClass().equals(clazz);
     }
   }
+
+  public static PTransformMatcher splittableParDoSingle() {
```
Comment (kennknowles) [inline runners/core-java/src/main/java/org/apache/beam/runners/core/PTransformMatchers.java]: javadoc

File: runners/core-java/src/main/java/org/apache/beam/runners/core/PTransformMatchers.java [hunk 2]
```diff
@@ -56,4 +60,64 @@ public boolean matches(AppliedPTransform<?, ?, ?> application) {
...
+          DoFn<?, ?> fn = ((ParDo.Bound<?, ?>) transform).getFn();
+          DoFnSignature signature = DoFnSignatures.signatureForDoFn(fn);
+          return signature.processElement().isSplittable();
+        }
+        return false;
+      }
+    };
+  }
+
+  public static PTransformMatcher stateOrTimerParDoSingle() {
```
Comment (kennknowles) [inline runners/core-java/src/main/java/org/apache/beam/runners/core/PTransformMatchers.java]: javadoc

File: runners/core-java/src/main/java/org/apache/beam/runners/core/PTransformMatchers.java [hunk 3]
```diff
@@ -56,4 +60,64 @@ public boolean matches(AppliedPTransform<?, ?, ?> application) {
...
+          DoFn<?, ?> fn = ((ParDo.Bound<?, ?>) transform).getFn();
+          DoFnSignature signature = DoFnSignatures.signatureForDoFn(fn);
+          return signature.usesState() || signature.usesTimers();
+        }
+        return false;
+      }
+    };
+  }
+
+  public static PTransformMatcher splittableParDoMulti() {
```
Comment (kennknowles) [inline runners/core-java/src/main/java/org/apache/beam/runners/core/PTransformMatchers.java]: javadoc

File: runners/core-java/src/main/java/org/apache/beam/runners/core/PTransformMatchers.java [hunk 4]
```diff
@@ -56,4 +60,64 @@ public boolean matches(AppliedPTransform<?, ?, ?> application) {
...
+          DoFn<?, ?> fn = ((ParDo.BoundMulti<?, ?>) transform).getFn();
+          DoFnSignature signature = DoFnSignatures.signatureForDoFn(fn);
+          return signature.processElement().isSplittable();
+        }
+        return false;
+      }
+    };
+  }
+
+  public static PTransformMatcher stateOrTimerParDoMulti() {
```
Comment (kennknowles) [inline runners/core-java/src/main/java/org/apache/beam/runners/core/PTransformMatchers.java]: javadoc

File: runners/core-java/src/main/java/org/apache/beam/runners/core/PTransformMatchers.java (hunk 1 above)
Comment (tgroh) [inline runners/core-java/src/main/java/org/apache/beam/runners/core/PTransformMatchers.java]: Done.

File: runners/core-java/src/main/java/org/apache/beam/runners/core/PTransformMatchers.java (hunk 2 above)
Comment (tgroh) [inline runners/core-java/src/main/java/org/apache/beam/runners/core/PTransformMatchers.java]: Done.

File: runners/core-java/src/main/java/org/apache/beam/runners/core/PTransformMatchers.java (hunk 3 above)
Comment (tgroh) [inline runners/core-java/src/main/java/org/apache/beam/runners/core/PTransformMatchers.java]: Done.

File: runners/core-java/src/main/java/org/apache/beam/runners/core/PTransformMatchers.java (hunk 4 above)
Comment (tgroh) [inline runners/core-java/src/main/java/org/apache/beam/runners/core/PTransformMatchers.java]: Done.

**Pull request closed without merging**